winrm4j/src/main/java/io/cloudsoft/winrm4j/winrm/WinRmTool.java

```diff
@@ -3,26 +3,111 @@
 import java.io.StringWriter;
 import java.nio.charset.Charset;
 import java.util.List;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
 
 import io.cloudsoft.winrm4j.client.WinRmClient;
-import io.cloudsoft.winrm4j.client.WinRmClient.Builder;
+import org.apache.http.client.config.AuthSchemes;
 
 /**
  * Tool for executing commands over WinRM.
  */
 public class WinRmTool {
+    private static final Logger LOG = Logger.getLogger(WinRmTool.class.getName());
+
+    public static final int DEFAULT_WINRM_PORT = 5985;
+    public static final int DEFAULT_WINRM_HTTPS_PORT = 5986;
+
     private String address;
     private String username;
     private String password;
+    private String authenticationScheme;
+    private boolean disableCertificatesChecks;
+
+    public static class Builder {
+        private String authenticationScheme = AuthSchemes.BASIC;
+        private Boolean useHttps;
+        private Integer port = null;
+        private boolean disableCertificatesChecks = false;
+        private String address;
+        private String username;
+        private String password;
+
+        private static final Pattern matchPort = Pattern.compile(".*:\\d+$");
+
+        public static Builder builder(String address, String username, String password) {
+            return new Builder(address, username, password);
+        }
+
+        private Builder(String address, String username, String password) {
+            this.address = address;
+            this.username = username;
+            this.password = password;
+        }
+
+        public Builder setAuthenticationScheme(String authenticationScheme) {
+            this.authenticationScheme = authenticationScheme;
+            return this;
+        }
+
+        public Builder disableCertificatesChecks(boolean disableCertificatesChecks) {
```

Comment:
`disableCertificatesChecks` will work for `NTLM` auth only, but `BASIC` will benefit form https even more - one more reason to use the async client for it.